### DISCUSSION TITLE
Re-enable automerge when PR is re-opened

### DISCUSSION THREAD

**Discussion by 0xSSA** [renovatebot/renovate#21163] [Suggest an Idea]
### What would you like Renovate to be able to do?

When Renovate reuses an existing branch, for example in the event that there is a newer version, Renovate closes the PR for the old version and re-opens it with the new version. However, after re-opening the PR, Renovate does not re-enable auto-merge. It could be nice if Renovate would re-enable automerge. 

### If you have any ideas on how this should be implemented, please tell us here.

Had a brief look at the source code. I see that there is already a check on whether a branch has been modified, within that check we could perhaps check if the PR has been re-opened. If that is the case, and the config is set to auto-merge, then re-enable auto-merge. 

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by rarkins** [comment]
Renovate should not be closing and reopening PRs when newer versions are available - it should update them in-place. Please double check and then provide more details about how you're running Renovate, which type of automerge, etc. Ideally next time please start with a Discussion and not directly to Issues like this

**Comment by 0xSSA** [comment]
Double checked. See provided screenshot for reference. Not re-enabling automerge, after re-opening closed PR. 

<img width="871" alt="Renovate" src="https://user-images.githubusercontent.com/119932026/227645391-0fb1a0c8-8fa1-4461-8cc9-f2d2438dcd00.png">

Using self-hosted Renovate, targeting Github hosted repositories with the following configuration: 

```
{
   "$schema":"https://docs.renovatebot.com/renovate-schema.json",
   "extends":[
      ":dependencyDashboard",
      ":semanticPrefixFixDepsChoreOthers",
      ":ignoreModulesAndTests",
      ":autodetectPinVersions",
      ":prHourlyLimit2",
      "group:monorepos",
      "group:recommended",
      "workarounds:all"
   ],
   "rangeStrategy":"bump",
   "packageRules":[
      {
         "matchUpdateTypes":[
            "minor",
            "patch"
         ],
         "automerge":true,
         "automergeType":"pr",
         "platformAutomerge":true
      }
   ]
}
```

**Comment by rarkins** [comment]
I'm not sure what was the reason for the autoclosing, but you can see it wasn't because "a new version was available" - both the old and new were same PR title.

I think this might be reproducible this way:
- Have a repo with an automerging PR
- Add a package rule to ignore that dependency = PR autoclosed
- Remove the package rule = PR reopened

**Comment by 0xSSA** [comment]
Thank you. Here is an additional screenshot that continues from the earlier screenshot. It does indeed seem like a new minor version was available. Renovate renames the PR title to the new version title after re-opening. Apologies for not providing that in the earlier screenshot. 

<img width="891" alt="Renovate 1" src="https://user-images.githubusercontent.com/119932026/227738764-98878469-4431-4195-b3e6-b5fcc5025d29.png">

**Comment by rarkins** [comment]
That doesn't mean that Renovate *closed* the PR because a new version was available. It does not do that. Please don't mix up the issues or confuse others

**Comment by 0xSSA** [comment]
I do not think I confuse anyone here. I am just reading the trail of events, which states that a PR was auto-closed and then reopened. Whereafter the auto-merge was not re-enabled. In this exact scenario, it just happened that a new minor version was available, but I do not think I state that I am certain that the auto-close of the PR is the direct result of that. 

However, the bottom line is that auto-merge is not re-enabled once a PR is auto-closed and then re-opened by Renovate.

**Reply by rarkins** [reply]
This statement is incorrect:

> in the event that there is a newer version, Renovate closes the PR for the old version

I've converted this into a discussion because it's now to the point where it will mislead future users searching issues. You can resubmit a clean feature request which does not contain the misleading statements.
